block/manager.go

```diff
@@ -164,27 +165,25 @@ func (m *Manager) Start(ctx context.Context) error {
 		<-m.DAClient.Synced()
 		nBytes := m.GetUnsubmittedBytes()
 		bytesProducedC := make(chan int)
-		go func() {
-			bytesProducedC <- nBytes
-		}()
-		eg.Go(func() error {
+		uerrors.ErrGroupGoLog(eg, m.logger, func() error {
 			return m.SubmitLoop(ctx, bytesProducedC)
 		})
-		eg.Go(func() error {
+		uerrors.ErrGroupGoLog(eg, m.logger, func() error {
+			bytesProducedC <- nBytes
 			return m.ProduceBlockLoop(ctx, bytesProducedC)
 		})
 	} else {
-		eg.Go(func() error {
+		uerrors.ErrGroupGoLog(eg, m.logger, func() error {
 			return m.RetrieveLoop(ctx)
 		})
-		eg.Go(func() error {
+		uerrors.ErrGroupGoLog(eg, m.logger, func() error {
 			return m.SyncToTargetHeightLoop(ctx)
 		})
 	}
 
 	go func() {
-		err := eg.Wait()
-		m.logger.Info("Block manager err group finished.", "err", err)
+		_ = eg.Wait() // errors are already logged
```

Comment:
so instead of logging the error here, it is now logged inside `uerrors.ErrGroupGoLog`?
I might be missing something, but I don't see how we benefit with this change.

Reply:
ah, nevermind. You wanted to log the error right away, not to wait for all the goroutines to end